### PR TITLE
expression: rewrite builtin function: CURDATE, SYSDATE, YEARWEEK

### PR DESCRIPTION
**Pull request by zz-jason**
to #4080

### CODE REVIEW

**Comment by XuHuaiyu** [comment]
ci failed, ptal

**Comment by zz-jason** [comment]
@XuHuaiyu fixed, PTAL.

**Comment by zz-jason** [comment]
@XuHuaiyu PTAL

**Comment by zz-jason** [comment]
fix #4245

**Comment by zz-jason** [comment]
@jackysp @breeswish PTAL

File: expression/builtin_time.go [hunk 1]
```diff
@@ -1049,49 +1051,83 @@ func (c *yearWeekFunctionClass) getFunction(args []Expression, ctx context.Conte
...
+}
+
+// evalInt evals YEARWEEK(date).
+// See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_yearweek
+func (b *builtinYearWeekWithoutModeSig) evalInt(row []types.Datum) (int64, bool, error) {
+	sc := b.ctx.GetSessionVars().StmtCtx
+
+	date, isNull, err := b.args[0].EvalTime(row, sc)
+	if err != nil {
+		sc.AppendWarning(err)
```
Comment (XuHuaiyu) [inline expression/builtin_time.go]: why always AppendWarnings here?

File: expression/builtin_time_test.go [hunk 2]
```diff
@@ -1313,7 +1318,6 @@ func (s *testEvaluatorSuite) TestUnixTimestamp(c *C) {
 	c.Assert(err, IsNil)
 	d, err = f.eval(nil)
 	c.Assert(err, IsNil)
-	c.Assert(d.GetInt64()-time.Now().Unix(), GreaterEqual, int64(-1))
```
Comment (XuHuaiyu) [inline expression/builtin_time_test.go]: why remove this?

File: expression/builtin_time.go (hunk 1 above)
Comment (zz-jason) [inline expression/builtin_time.go]: a mistake

**Comment by zz-jason** [comment]
@XuHuaiyu PTAL

**Comment by XuHuaiyu** [comment]
LGTM

**Comment by breeswish** [comment]
LGTM

**Pull request merged**